client/onboarding/index.tsx

```diff
@@ -46,13 +46,27 @@ const OnboardingStepper = () => {
 	);
 };
 
+const getComingSoonShareKey = () => {
+	const {
+		woocommerce_share_key: shareKey,
+		woocommerce_coming_soon: comingSoon,
+		woocommerce_private_link: privateLink,
+	} = wcSettings?.admin?.siteVisibilitySettings || {};
+
+	if ( comingSoon !== 'yes' || privateLink === 'no' ) {
+		return '';
+	}
+
+	return shareKey ? '/?woo-share=' + shareKey : '';
```

Comment:
@psealock after getting help in Slack and trying with 9.0.0-beta I was able to test it!
There is one small thing
The share link was is `https://meerkat-of-crawdads.jurassic.ninja/?woo-share=iYjAGpd9yKwDKRRsB85MzQKdWPIOg7tQ`
And the link passed in `self_assestment` had extra slash `https://meerkat-of-crawdads.jurassic.ninja//?woo-share=iYjAGpd9yKwDKRRsB85MzQKdWPIOg7tQ`
```suggestion
	return shareKey ? '?woo-share=' + shareKey : '';
```